### PR TITLE
Porting 'stm32fcube' bsp drivers

### PR DESCRIPTION
**Pull request by anton-bondarev**
1. Use stm32fXcube bsp for uart and ethernet drivers
2. Use stm32fXcube for f3, f4 and f7 discovery boards
3. Enable several ttyS devices for stm32f4 board

### CODE REVIEW

File: third-party/bsp/stmf7cube/cmsis_impl.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ * @brief
+ *
+ * @author  Anton Kozlov
+ * @date    12.08.2014
+ */
+
+#ifndef ST_F4_CMSIS_IMPL_H_
+#define ST_F4_CMSIS_IMPL_H_
+
+//#include <stm32f4xx.h>
```
Comment (Deryugin) [inline third-party/bsp/stmf7cube/cmsis_impl.h]: Remove this comment please

File: third-party/bsp/stmf7cube/Makefile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+PKG_NAME := stm32f7
+
+PKG_SOURCES := http://www.st.com/st-web-ui/static/active/en/st_prod_software_internet/resource/technical/software/firmware/stm32cubef7.zip
+PKG_MD5     := 7e5bfa00de6d77c86c25371420902d32
+
+#PKG_PATCHES := buildfix.patch
```
Comment (Deryugin) [inline third-party/bsp/stmf7cube/Makefile]: One more unnecessary line

File: third-party/bsp/stmf4cube/Makefile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+PKG_NAME := stm32f4
+
+PKG_SOURCES := http://www.st.com/st-web-ui/static/active/en/st_prod_software_internet/resource/technical/software/firmware/stm32cubef4.zip
+PKG_MD5     := 162a5b53368cacf4f7f577a45ee8d282
+
+#PKG_PATCHES := buildfix.patch
```
Comment (Deryugin) [inline third-party/bsp/stmf4cube/Makefile]: One more unnecessary line

**Comment by Deryugin** [comment]
Well, the code seems ok, but there are big chunks of code taken from stm bsp. It will take some time to understand what it actually does.

Does it work for different boards, or just for stm32f4?

**Comment by anton-bondarev** [comment]
Thank for comments
In a theory this code can be use for all stm boards (f3/f4/f7) board, but really it was checked only on stm32f4-discovery.

**Comment by Deryugin** [comment]
Well, we will not have other boards for another week, so we can merge it :)

**Pull request merged**